src/sentry/static/sentry/app/views/eventsV2/landing.tsx

```diff
@@ -64,29 +65,58 @@ class DiscoverLanding extends AsyncComponent<Props, State> {
     errors: [],
     isBannerHidden: checkIsBannerHidden(),
     savedQueries: [],
+    savedQueriesPageLinks: '',
   };
 
   shouldReload = true;
 
   getEndpoints(): [string, string, any][] {
-    const {organization} = this.props;
+    const {organization, location} = this.props;
+    const views = getPrebuiltQueries(organization);
+    const cursor = location.query.cursor;
+    // XXX(mark) Pagination here is a bit wonky as we include the pre-built queries
```

Comment:
So, I've run into this bug. 🤔

Reply:
Is there a more sufficient fix for this?

Reply:
Possibly but it will need a custom paginator handler similar to the one in your pull request.